base/applications/cmdutils/tasklist/tasklist.c

```diff
@@ -0,0 +1,398 @@
+        if (ProcessInfoBuffer)
+        {
+            PBYTE NewProcessInfoBuffer = HeapReAlloc(GetProcessHeap(), 0, ProcessInfoBuffer, ProcessInfoBufferLength);
+            if (NewProcessInfoBuffer)
+            {
+                ProcessInfoBuffer = NewProcessInfoBuffer;
+            }
+            else
+            {
+                // out of memory ?
+                ConResMsgPrintf(StdOut, 0, IDS_OUT_OF_MEMORY);
+                HeapFree(GetProcessHeap(), 0, ProcessInfoBuffer);
+                FreeLibrary(hNtDLL);
+                return FALSE;
+            }
+        }
+        else
+        {
+            ProcessInfoBuffer = HeapAlloc(GetProcessHeap(), 0, ProcessInfoBufferLength);
+            if (!ProcessInfoBuffer)
+            {
+                // out of memory ?
+                ConResMsgPrintf(StdOut, 0, IDS_OUT_OF_MEMORY);
+                FreeLibrary(hNtDLL);
+                return FALSE;
+            }
+        }
+
+        // Query information
+        Status = PtrNtQuerySystemInformation(
+            SystemProcessInformation, ProcessInfoBuffer, ProcessInfoBufferLength, &ResultLength);
+
+        if (Status != STATUS_INFO_LENGTH_MISMATCH)
+        {
+            break;
+        }
+    }
+
+    if (!NT_SUCCESS(Status))
+    {
+        // tried NT_SYSTEM_QUERY_MAX_RETRY times, or failed with some other reason
+        ConResMsgPrintf(StdOut, 0, IDS_ENUM_FAILED);
+        HeapFree(GetProcessHeap(), 0, ProcessInfoBuffer);
+        FreeLibrary(hNtDLL);
+        return FALSE;
+    }
+
+    // print header
+    PrintHeader();
+
+    PSYSTEM_PROCESS_INFORMATION pSPI;
+    pSPI = (PSYSTEM_PROCESS_INFORMATION)ProcessInfoBuffer;
+    while (pSPI)
+    {
+        PrintString(pSPI->UniqueProcessId ? pSPI->ImageName.Buffer : L"System Idle Process", COLUMNWIDTH_IMAGENAME);
+        PrintSpace(1);
+        PrintNum((long long)(long)pSPI->UniqueProcessId, COLUMNWIDTH_PID);
+        PrintSpace(1);
+        PrintNum((long long)pSPI->SessionId, COLUMNWIDTH_SESSION);
+        PrintSpace(1);
+        PrintMemory(pSPI->WorkingSetSize, COLUMNWIDTH_MEMUSAGE, 1);
+
+        ConPrintf(StdOut, L"\n");
+
+        if (pSPI->NextEntryOffset == 0)
+            break;
+        pSPI = (PSYSTEM_PROCESS_INFORMATION)((LPBYTE)pSPI + pSPI->NextEntryOffset);
+    }
+
+    HeapFree(GetProcessHeap(), 0, ProcessInfoBuffer);
+    return TRUE;
+}
+
+int GetArgumentType(WCHAR* argument)
+{
+    int i;
+
+    if (argument[0] != L'/' && argument[0] != L'-')
+    {
+        return OP_PARAM_INVALID;
+    }
+    argument++;
+
+    for (i = 0; i < _countof(opList); i++)
+    {
+        if (!wcsicmp(opList[i], argument))
+        {
+            return i;
+        }
+    }
+    return OP_PARAM_INVALID;
+}
+
+BOOL ProcessArguments(int argc, WCHAR *argv[])
+{
+    int i;
+    BOOL bHasHelp = FALSE, bHasVerbose = FALSE;
+    for (i = 1; i < argc; i++)
+    {
+        int Argument = GetArgumentType(argv[i]);
+
+        switch (Argument)
+        {
+        case OP_PARAM_HELP:
```

Comment:
Missing space for all the case bodies here within the switch.

Reply:
fixed now, thanks 🤣